Shuttle (New): NC Resolve

goldkin opened the pull request:
## About the PR
Adds a new medical ship: the NC Resolve, as well as its supporting documentation.

## Why / Balance
### Why
To give another option than the Stasis for learning how cryopods work and when running small-crew rescue operations in deep space.

The Stasis is a fantastic ship design (you did great, @dustylens!), and it fills its niche well. Too well. So well that basically everyone buys one if they want to run their own cryopod and medical chems lab, since it's an affordable full-service option for paramedic duties and for triaging emergency medical pods on quieter shifts.

The other ship-based option for cryo at this writing -- other than building your own during the round -- is the Caduceus. This is also a fantastic ship, but since it's expensive and fairly large for small crews, it doesn't show up quite as often (on Hypatia and Maunder, at least).

The other options in the medical roster work well for, eg, rapid response, chems, light paramedic duties, and support roles (among other notables). But with those, the lack of cryo prevents them from accessibly being able to treat all patients that come in. It's possible to use this as a triage opportunity to get assistance from other ships, but typically this means supporting someone else already running a Stasis. It's also technically possible to treat these patients by spacing them in an appropriate suit and performing "ghetto chem", but this requires advanced knowledge of the game's mechanics and limits roleplay opportunities when helping out other players.

Given all that, I feel there's a gap in the medical roster that this ship fills.

Note also: this is expressly _not_ to provide a cheaper option for soloing medical pods. The Stasis is already used this way in certain cases, and this ship is only slightly cheaper (48000 vs 56000; 40000 appraisal + 20% markup) for fewer facilities. The goal is to break up the monotony of one ship design being best suited for small crews that want to run full-service medical, and to give an alternative layout with its own unique feel and roleplay opportunities.

### General design considerations and balance

This ship is intended as a tier 2 (full medical, half roleplay/service) medium ship. The layout is 230 tiles at 19w x 17h.

The design of this ship is inspired by local, real medical wards (which I have, unfortunately, been to a lot recently to assist ailing family) crossed with "building a recognizable silhouette" that can be identified immediately as medical. I went with a plus-sign shaped design, which lends itself well to having a reception hallway and all other services radiating from it.

Notable features and design notes:
* Certain areas of the ship are set up with a more roleplay focus in mind for those so inclined, or extra storage and desire paths if not.
* I tried to make the cryo loop simple to observe on t-ray scanning, to make it easier to learn how to engineer one's own cryo from this setup. This meant displacing some of the atmo pipes to accommodate this, and I think the current layout works well. 
* I went a bit beyond the style guide to try to fit certain other building principles, including fire codes. This mostly dictates where the fire extinguisher cabinet is, why there's an emergency EVA by the window in medical, and why each room has pedantic levels of adequate ventilation and fire safety (other than the separator between hydroponics and medical). It's somewhat difficult to break reinforced shuttle windows with the tools on hand, so not sure how well this will work in practice.
* There are several desire paths deliberately built into this design for plucky engineers, all with interesting tradeoffs. Everything should be relatively easy to upgrade. Removing the splash guards in chemistry, turning the windoors between medical and hydro into a walkable path, and clearing out the morgue and front desk to allow for more cargo storage are all deliberate and expected for players to try, to their own (mis)fortune should any chemicals, gasses, or kudzu (etc) spill or detonate from these locations.
* This uses plasma (x2) for its engines, a ChemVend for extra chemicals storage, and deliberately lacks a NanoMed. This is mostly to condense things given the small ship design, but there's plenty of space to revisit this if needed.
* This uses the NanoTransen Civilian ship manufacturer, which is typically for service vessels. I thought it would be neat for lore reasons to have them sponsor this ship (this is further expanded upon in the guidebook page). I tried to fit some of the design close to these ships to make it feel this way, and chose 20% markup because that's what the Bocadillo uses while fitting into tier 2 pricing.

## Technical details
This PR includes:
1. The ship itself and supporting metadata
2. A guidebook entry and supporting media (read: a 128x96 png entry for the guidebook, kitbashed by manual pixel editing and pulling the text in from existing checked-in guidebook pages using GIMP)
3. Credits in the places that need them

The following ship components cannot be unanchored, to prevent "high-grading," buying the ship to bypass certain research objectives, and as general gameplay balance considerations:
1. Freezer
2. ChemVend
3. All three hydroponics trays

The following ship components cannot be unanchored to prevent griefing (ie, the kind that would be unfun and unsuited to constructive roleplay through antag roles):
1. Withdraw-only bank ATM (wall-mounted). This entity appears to have other protections preventing unanchoring at the server side, but I also set it as unanchorable on the component itself.

This is my first PR for the Frontier fork and for SS14 generally, and while I tried to follow all the things including the styleguide (new and old), please let me know which parts I undoubtedly missed. 💜 

Special thanks to Eb K'eth for many many many suggestions and design reviews, and to teybob for suggestions that allowed the current design to be tidy and compact.

## How to test
1. Launch NFDev (in release mode!), spawn a medical shipyard console, buy the ship
2. For the guidebook entry, you can find the relevant entry in the guidebook (Shipyard > Resolve) or in the manual tab at purchase (no in-game book object for it, yet)

Things I tested in no particular order follow. Adding media for all of these is possibly overkill, so using prose for now. Let me know if you'd like to see specific cases:
* Sight lines and general look and feel from the player's perspective
* Rebuilding the YAML to ensure there aren't any serialized locker fills, previously initialized components, and dangling log or forensics data from development (door access logs, etc)
* Networked components (vents, scrubbers, firelocks, air alarms, and blast door-button linking)
* Entities drawing power from the power network
* Locker and device accessibility tests (read: can I reach and use all of them, does doing so cause other problems like glass tables shattering under me, etc)
* Fax machine and bluespace holopad calls
* Anchoring and unanchoring components (including verifying unanchorable components)
* Airlocks and atmospherics, including defaults when the ship spawns
* Power down and power on tests at the APC and generator level
* Variantize tests (manually; `variantize` does not work for me for some reason?)
* Firelock activation (automatic and manual)
* Fire and chemical isolation tests (read: are these contained as expected by walls and doors during accidents?)
* Filling the ship full of monkeys, space fleas, and space kangaroos
* Blowing the ship up in various creative and elaborate ways to see what happens (Did you know that morgue units generally survive large explosions, but their contents do not? News to me!)
* Screaming into the void
* Realizing just how many moving parts and considerations go into a complete ship design, and how rewarding that is
* Iterative bugfixes on my end condensed into this single PR for simplicity

What I did not test:
* Upgrading components using science tier unlocks or other upgrade paths (eg, finding and installing RTGs).
* A full simulated medical shift on the current version. I did simulate this using emergency medical pods on an earlier version, though.
* Whether the placement of the door to Medical is in a good spot for crews bringing in patients and cargo. I think adding a second glass door may help here, and I'd be interested in playtest feedback from other players to confirm!

## Media
Mapping view of the shuttle:
![Screenshot 2025-03-20 011111](https://github.com/user-attachments/assets/3ac4cd91-695f-4eae-b9bd-e87e3c323238)

Lighted version (mapping and POV disabled):
![Screenshot 2025-03-20 011536](https://github.com/user-attachments/assets/4766abf6-06a9-468d-9214-81f6d3e1c893)

Emergency lighting:
![Screenshot 2025-03-20 002245](https://github.com/user-attachments/assets/2b961fca-e34c-4c13-a9af-713072dbbcd4)

Crew monitor view:
![Screenshot 2025-03-18 172456](https://github.com/user-attachments/assets/ac7e19a6-4e48-428d-985d-dfe0a33b2a1a)

Mass scanner signature:
![Screenshot 2025-03-18 172544](https://github.com/user-attachments/assets/1eea2510-8e7f-47ae-92a4-945637ae686a)

APC loads are documented in the manual. These should be 9.2 - 14.2 kW for the upper APC (depending on cryo) and 8.1 kW for the lower without adjustments.

## Requirements
- [X] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
- [X] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [X] I confirm that AI tools were not used in generating the material in this PR.

## Breaking changes
None I'm aware of.

**Changelog**
This review:
- add: NC Resolve introduced.
- add: NC Resolve manual pages added.

Eb- commented:
I'm very happy to see this PR submitted! It's been a pleasure reviewing the ship's progress throughout the build stages, and I hope to see it be play-tested in game, and eventually accepted into the shipyard consoles. You've put a lot of thought into the ship and it shows!

goldkin commented:
@dustylens Thanks for having a look! I'll address particular points in a rework, and publish the diff downstream so we can compare the results.

Generally, I tried to be _too_ sparse with tile differentiation because of general guidance for it to fall to the background compared to foreground elements. I agree that there's a lot I can do there to liven it up -- and I'll integrate more of those ideas.

Regarding generally condensing spaces too much: substantially agree. Some considerations you called out are for compactness, but some (notably where the battery charger is!) are more owed to be being frustrated on vessels where I can't figure out which farking room the charger is placed in. That said, I can probably make this easier to find than stuffing it in the main hallway.

Re, ants on the bed: lol. I think I went overboard on not filling the storage immediately next to it to prevent serialization. I'll drop it there with storage open or delete it.

The sunflower seeds and blue/yellow bedsheets were inspired as a very understated nod in support of Ukraine. I want to keep it as a design note, but I can place it better without issue.

The other puzzling elements are probably cases where I overextended too much on specific ideas. Why a book at reception for example: roleplay for a person sitting there doing paperwork. I can leave this entirely absent -- folks who want to roleplay can bring their own Fun(tm).

Overall, what I care about the most is how folks _use_ the space, and I hope that shows and works generally well. I'll leave the rest for a rework candidate, and we'll see where it goes. Appreciated!

goldkin commented:
> If you reach a point where you would like to discuss ideas please feel free to raise a flag!

@dustylens Turns out that timing is right now, because one of our cats had the zoomies and woke me after sleeping on this feedback.

Here's a very early rework WIP to get your feedback on the layout, specifically. Because I _think_ I can preserve the general character of the shuttle while integrating most of what you mentioned:
![Screenshot 2025-03-22 072209](https://github.com/user-attachments/assets/8d8b6464-3f7d-4a68-8b51-91433dfcf0df)

I was thinking of trying more radical reworks, and honestly, that might be in the cards for some of the rooms. But I kind of like the current plan (inclusive of how it feels to move around the space and play), and want to preserve it if I can.

Generally, I increased the overall dimensions of the shuttle by 2 tiles each. I used the extra space to grow each room that benefitted from it.

Individual design points follow:
### Airlock
Not too much to change here. I intend to do a general decal pass (probably not as pedantic as, say, caution on _every_ airlock like the gasbender, but something consistent).

I added your signage suggestions to the external airlock side. Had completely missed how useful those are on other shuttles, thanks!

### Lobby
No more book spawner. Kept the pink shark (pink is generally the color for cancer survivors, and this is exactly the kind of minimal addition that gives a lot of character to a shuttle without being too imposing, IMO).

This rework will happen last, after the other rooms are mostly complete. I think moving reception slightly and adding a small nook for the blast door will fix most of the... shall we say, feel that problem customers get spaced. I still think this is generally the right zone for cargo loading though, so I'll noodle with this and see what happens.

I don't feel too strongly about where the devices sit right now. I expect to rejigger these between the lobby and the medical bay at some point.

### Morgue
Okay, so this largely got skipped, but I actually have strong feelings on this one.

I really, _really_ want to give folks the ability to use this for detective roleplay, and for generally feeling as though morgues have a place in the current game (instead of being convenient stash points with reduced rot damage, eventually leading to a revival or simply forgetting bodies are there).

The whole X-Files dynamic of Skully patiently explaining to a bored Mulder (and by extension, the audience) what fresh hell had happened to their patient fits detectiving very well, and I passably want to give this the ability to serve this function.

_That said_, realistically this is going to end up as cargo storage like 95% of the time. The Stasis has the same problem -- in fact, I had no idea that chems storage was the morgue until I read its documentation for this project!

By giving this about 6 more tiles, my general plan is to move the desk setup down and let this get used as a bonus storage bay (for materials and for bodies), until the need strikes to use it as a detective space or for paperwork. I'm against also using this as spare chemical storage like the Stasis. But, if I feel the ChemVend doesn't serve this well enough, this might grow a chemicals crate or two.

### Engineering
Surprisingly not much here. I had a hard time fitting a Super P.A.C.M.A.N in here because uranium demands a proper blast door setup (compared to plasma) to be consistent with existing guidance. Adding six tiles to the room lets me do that and gives just enough room for an SMES (to be added).

### Dorms
This is going to get slightly more living quarters with 6 more tiles. I need to rearrange where the windows are, but I have a lot to work with here now.

The glass door was so folks could easily see that it exists (same as why hydro has one, it lets players clearly see the Bridge sign as they pass by. But this room is less essential, and I think folks will bump into it while exploring the ship for the first time.

Also, thanks for the note on the CMO locker no longer being a thing. This is news to both me and @Eb-, and I'll try to hunt down the Stasis PR to see if anything other than another cabinet should be added in its place.

### Hydro
So one of the things I feel really strongly about on this design is not overwhelming the player. I have doors and sight lines set up kind of explicitly so you see only a couple rooms at most angles, absent a few extra views out into space.

The problem I had with hydro was when walking through medical, you'd get sight out windows in both directions. This made it a bit overwhelming to work medical, since that extra view would extremely distract me (and presumably, other players as well).

At the same time, I didn't want the player in hydro to feel isolated and trapped during their shift. @Eb- mentioned that players really like their views out windows (it's true!), so to solve this problem, I added the glass under each hydro tray to make it _feel_ like you were looking out into space when you weren't.

Anyway, by increasing the room size this problem goes away entirely. I can move the window up a bit so it's no longer in line with the windoor/grinder setup (while not making it look terrible) and turn these back into some variation on plating.

### Medical Bay
This benefits the most from the increased amount of space, and honestly, resizing this let me fix a lot of problems it has.

I'm not entirely certain if chem will be able to get its own separate room out of this setup, but the walls and overall room setup feels much better now. This also lets me move its air sensor down to a more accessible and visible location, the emergency light to a wall instead of a window, and gives more space for my atmo rework when I get to it.

As for the posters: they're gone for now. I'd been using legit random poster spawner instead of the general one, and variations on safety moth showing up here made the space feel good to work in. I don't want to select _specific_ posters or go so far as to say, write my own random poster spawner, but this needs _something_.

Anyway, right now this room is best viewed as a rough draft, but I think I'm satisfied with how it'll come together.

### Gas systems
I don't think combining the waste loop is a tall order at all. I'd ideally like to avoid overlapping pipes, but this is a topological problem that I can solve by nesting one loop inside of the other to do that. If that makes sense.

I think I can preserve the simplicity of the cryo piping nested inside of both of those. Leaving this out for now, until I can sit down and spend 30m to an hour playing the design equivalent of Pipe Dream.

### Thruster systems
Not much here. I haven't benchmarked how other shuttles behave, and that should clear most of this up.

More thruster gives more zoomies. Plenty of space to add them now and playtest to satisfaction.

### Decorations, decals and tiling
Largely untouched for now. Though I did notice I was putting all of my windows on lattice instead of plating and fixed that.

This will track late, but I want to do something consistent for the airlocks as above. Otherwise, trying to not overdo it so folks can build their own stories into the shuttle, but I'll go a bit harder on the tiling.

### Things that didn't come up
Honestly, I was expecting my engineering wireplan would draw more feedback and required changes. The fact it didn't (beyond changing the generation strategy, adding an SMES, etc) likely means I did this somewhat coherently, and I'll keep most of where the wires go.

It also sounds like I did the right thing with most of my variables and the guidebook setup, insofar as it works as expected and didn't violently explode in your own NFDev. Do let me know if there's changes that should happen there, other than switching plasma to uranium, the map when the rework is done, and some prose updates here and there to accommodate these changes.

Anyway, that's all I've got for now. Let me know if any of this sounds radically off-base, otherwise my plan is to keep refining things from here. Thanks!

goldkin commented:
So it took a couple hours of messing with ideas and playing elaborate Pipe Dream with atmos, but I think I arrived at something workable:
![Screenshot 2025-03-23 074336](https://github.com/user-attachments/assets/901c927b-c945-4422-b0cf-414838c548e6)
![Screenshot 2025-03-23 074424](https://github.com/user-attachments/assets/d107fd5f-4267-4ec8-8fe7-64d6649363f7)

One of the big improvements here is building in many of the desire paths that weren't there in v1:
* Cargo now has a path right to the morgue, for extra storage
* Hydro now has a proper door, and a path directly from plants and the grinder to the chemmaster
* Medical is a lot more open, and the cryo and beds setup are now condensed to make it easier to serve patients
* Larger door setups so folks don't get blocked moving cargo or pods (etc)

I also took the opportunity to remind myself which recipes ended up where, and condensed devices by function. I decided to turn the path to engineering into a useful place to assemble meds, by putting the assembler right by both the chemvend and the (now added) nanomed plus. This now serves as both extra chems storage and a self-service kiosk for crafting meds to go.

There's a ton of other things that changed, and this isn't done -- I still need to rewire and rework the thrusters. So no yml upload just yet. But I wanted to share my draft of where this was going to see if it's in the general ballpark of where folks want it to go!

goldkin commented:
Prototype YAML is in. Here's what the shuttle looks like now:
![Screenshot 2025-03-23 232347](https://github.com/user-attachments/assets/57904a2a-b9a2-46bc-9570-d62b8ce603e0)
![Screenshot 2025-03-23 232910](https://github.com/user-attachments/assets/f87c5b74-82b9-4668-84e2-36d1846a7680)
![Screenshot 2025-03-23 222743](https://github.com/user-attachments/assets/e4fa60b9-6c1c-445e-8513-bf4cab9ad90d)
![Screenshot 2025-03-23 222824](https://github.com/user-attachments/assets/8bdfe474-e351-426d-9725-a0fa216a111e)

Definitely a step up compared to my original submission. Let me know what you think!

goldkin commented:
Three of us (myself, @Eb- and their partner) gave this a test run in a simulated two hour shift to see how it fared. I made some adjustments accordingly that are now on this branch.

I'm too sleepy right now to add more than a few notes and some screenshots, but this is feeling like a viable design now:
![Screenshot 2025-04-07 022636](https://github.com/user-attachments/assets/59b8a445-d487-43c5-bee9-c2f929fbfd8a)
![Screenshot 2025-04-07 023207](https://github.com/user-attachments/assets/44556f38-766a-4251-80f0-3b925fc43cc9)

Feel free to give it a try and see what you think.

goldkin commented:
Docs should now be up to date with the current state of the ship. Will leave this open for the time being.

If there's interest, feel free to give it a test! If not, I'm satisfied to keep it on my own fork for now until it's wanted. Thanks!

whatston3 commented on Resources/Prototypes/_NF/Shipyard/resolve.yml:
```diff
@@ -0,0 +1,44 @@
+          prefixCreator: '14'
+      - type: StationJobs
+        availableJobs:
+          Contractor: [ 0, 0 ]
+          Pilot: [ 0, 0 ]
+          Mercenary: [ 0, 0 ]
```
```suggestion
          ContractorInterview: [ 0, 0 ]
          PilotInterview: [ 0, 0 ]
          MercenaryInterview: [ 0, 0 ]
```
Use new interview jobs.

goldkin commented:
> Hi Goldkin, are you still developing this PR?

Kinda. This design is as far along as I think I plan to take it (barring conflicts and the review from 2wks ago, which I'll merge).

I think we'll want to make a decision on whether to accept or reject it for now. I'm favoring "reject", because there really hasn't been much appetite for this design outside of our small testing group right now, and since I posted this at least one other option with cryo have shipped to main.

So I'm thinking: it was super rewarding to build this, but if there's no interest, that's enough evidence to archive it for now. The PR and branch will be there on my fork should anyone want it.

goldkin commented:
Also, the Retort is close enough of a similar name and role that I feel that the Resolve would cause some confusion.

So if we do end up accepting this (see above), I'd be interested in a gut check on that and some alternative name suggestions.
